conjure-runtime/src/rt/js.rs

```diff
@@ -0,0 +1,99 @@
+// Copyright 2025 Palantir Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+pub mod time {
+    use std::future::Future;
+    use std::ops::{Add, Sub};
+    use std::pin::Pin;
+    use std::task::{ready, Context, Poll};
+    use std::time::Duration;
+    use tokio::sync::oneshot;
+    use wasm_bindgen::prelude::{wasm_bindgen, Closure, JsValue};
+    use web_sys::Performance;
+
+    #[wasm_bindgen]
+    extern "C" {
+        #[wasm_bindgen(js_name = "performance", thread_local_v2)]
+        static PERFORMANCE: Performance;
+        #[wasm_bindgen(js_name = "setTimeout")]
+        fn set_timeout(cb: &Closure<dyn FnMut()>, millis: f64) -> JsValue;
+        #[wasm_bindgen(js_name = "clearTimeout")]
+        fn clear_timeout(timeout_id: &JsValue);
```

Comment:
In theory pulling these from the global context instead of the Window will allow this to work on a WebWorker as well, but I haven't tested that.